DAOS-14491: Retain support for phase-1 DAV heap

sherintg opened the pull request:
A copy of the phase-1 DAV allocator with some cleanup is moved to the subdirectory src/common/dav_v1. This allocator is built as a standalone shared library and linked to the libdaos_common_pmem library.  The umem will now support one more mode DAOS_MD_BMEM_V1 which results in umem instance using phase-1 DAV allocator interface and layout.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/1/execution/node/331/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/1/execution/node/365/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/1/execution/node/344/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/2/execution/node/379/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/2/execution/node/354/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 9 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/3/execution/node/368/log

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/3/execution/node/378/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/3/execution/node/361/log

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/3/execution/node/375/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build RPM on EL 8 completed with status UNSTABLE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/4/execution/node/353/log

daosbuild1 commented:
Test stage Build RPM on Leap 15.4 completed with status UNSTABLE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/4/execution/node/386/log

brianjmurrell commented on utils/rpms/daos.spec:
I understand this is on a feature branch, but just keep in mind that any update to this file that results in a packaging difference needs to include a `Version:` or `Release:` bump and a `%changelog` entry.

You probably also need to update `debian/` to account for this new file and update `debian/changelog` accordingly.

sherintg replied:
thanks @brianjmurrell. I will update them too. Is there a documentation that I can refer to?

brianjmurrell replied:
Probably just looking at the history of the individual files (https://github.com/daos-stack/daos/commits/master/utils/rpms/daos.spec, https://github.com/daos-stack/daos/commits/master/debian/changelog, https://github.com/daos-stack/daos/commits/master/debian/control) will tell you all you need to know.  You should be able to find quite easily a commit that updates all three due to adding a library (or any other file).

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 commented:
Test stage Build DEB on Ubuntu 20.04 completed with status FAILURE.  https://build.hpdd.intel.com//job/daos-stack/job/daos/view/change-requests/job/PR-13158/7/execution/node/365/log

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

brianjmurrell commented on src/common/dav/bucket.c:
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2022, Intel Corporation */
```
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```
yes?

brianjmurrell commented on src/common/dav/bucket.h:
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2021, Intel Corporation */
```
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```

brianjmurrell commented on src/common/dav/dav.h:
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2022, Intel Corporation */
```
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```

brianjmurrell commented on src/common/dav/heap.c:
```diff
@@ -1,5 +1,5 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
-/* Copyright 2015-2023, Intel Corporation */
+/* Copyright 2015-2022, Intel Corporation */
```
```suggestion
/* Copyright 2015-2023, Intel Corporation */
```

I won't comment on any more erroneous copyright date changes but please review all of your changes for this and correct.

brianjmurrell commented on utils/rpms/daos.spec:
```diff
@@ -585,6 +586,10 @@ getent passwd daos_agent >/dev/null || useradd -s /sbin/nologin -r -g daos_agent
 # No files in a shim package
 
 %changelog
+* Wed Oct 16 2023 Sherin T George <sherin-t.george@hpe.com> 2.5.100-10
```
```suggestion
* Mon Oct 16 2023 Sherin T George <sherin-t.george@hpe.com> 2.5.100-10
```
perhaps?

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.

sherintg commented:
@brianjmurrell I have addressed your review comments. Apologies for making multiple force pulls after your initial review comments. There was a major design change requested and I had to redo the entire coding from the beginning.

brianjmurrell reviewed (commented):
Packaging changes look good.  I will leave +1 slots for those more familiar with the bulk of the changes in this PR.

NiuYawei commented on Jenkinsfile:
```diff
@@ -956,7 +956,7 @@ pipeline {
                     steps {
                         job_step_update(
-                            unitTest(timeout_time: 60,
+                            unitTest(timeout_time: 120,
```
What's the change for?

daosbuild1 reviewed (commented):
LGTM.  No errors found by checkpatch.
